javaparser-core/src/main/javacc/java.jj

```diff
@@ -711,36 +711,46 @@ StubUnit StubUnit():
     ModifierHolder modifier;
     TypeDeclaration tn = null;
     ModuleDeclaration module = null;
+    NodeList<AnnotationExpr> annotations = new NodeList<AnnotationExpr>();
+    AnnotationExpr annotation = null;
 }
 {
     ( LOOKAHEAD(2) ";" )*
-    [ LOOKAHEAD(PackageDeclaration()) pakage = PackageDeclaration() ]
+    /*
+     * Reading leading imports to support the old stub files that contain import declarations
+     * before the package declaration.
+     */
+    (in = ImportDeclaration() { imports = add(imports, in); })*
+
     /**
      * The stub file format permits having multiple classes and packages.
      * The packages are separated by a package statement: package my.package;.
      * Each package declaration may occur only once; in other words, all classes from a package must appear together.
      *
      * Going through a stub file and reading one compilation unit at a time
      * Number of compilation units in a stub file is 1+
-     */
-    (
-        /* Additional package declaration is not require, but still possible */
-        (pakage = PackageDeclaration())?
-
-        /*
-         * The only required import statements are the ones to import type annotations.
-         * Import statements for types are optional.
-         */
-        (in = ImportDeclaration() { imports = add(imports, in); })*
+     * Each CompilationUnit declaration starts with a package declaration.
+    */
+    (pakage = PackageDeclaration())?
+    (in = ImportDeclaration() { imports = add(imports, in); })*
 
-        {
-            types = emptyList();
-            modifier = null;
-            tn = null;
-            module = null;
-        }
-
-            (
+    (LOOKAHEAD(3, PackageDeclaration())
+        (
+             {
+                cu = new CompilationUnit(range(token_source.getHomeToken(), token()), pakage, imports, types, module);
+                compilationUnits.add(cu);
+                types = emptyList();
+                tn = null;
+                module = null;
+             }
+             pakage = PackageDeclaration()
+             (in = ImportDeclaration() { imports = add(imports, in); })*
```

Comment:
I understand that.  I'm asking what happens if the combined list contains conflicts that none of the individual lists would have.
(Maybe nothing bad happens?  Or maybe it does, and there should be a global list for imports before the first `package` and there should be a local list for each compilation unit, and a compilation unit gets the concatenation of those two rather than the concatenation of every import that precedes it in the file.)